src/main/java/com/structures/helpers/Structure.java

```diff
@@ -0,0 +1,422 @@
+
+            for (BlockRenderLayer layer : BlockRenderLayer.values())
+            {
+                if (block.canRenderInLayer(actualState, layer))
+                {
+                    this.mc.getTextureManager().bindTexture(TextureMap.LOCATION_BLOCKS_TEXTURE);
+                    ForgeHooksClient.setRenderLayer(layer);
+                    this.renderGhostBlock(world, holder, player, layer, existingModel, partialTicks);
+                    holder.rendered = true;
+                }
+            }
+
+            ForgeHooksClient.setRenderLayer(originalLayer);
+        }
+
+        if (holder.te != null && !holder.rendered)
+        {
+            TileEntity te = holder.te;
+            te.setPos(holder.pos);
+            FakeWorld fakeWorld = new FakeWorld(holder.actualState, world.getSaveHandler(), world.getWorldInfo(), world.provider, world.theProfiler, true);
+            te.setWorldObj(fakeWorld);
+            int pass = 0;
+
+            if (te.shouldRenderInPass(pass))
+            {
+                TileEntityRendererDispatcher terd = TileEntityRendererDispatcher.instance;
+                terd.func_190056_a(fakeWorld,
+                        Minecraft.getMinecraft().renderEngine,
+                        Minecraft.getMinecraft().fontRendererObj,
+                        new FakeEntity(fakeWorld),
+                        (RayTraceResult) null,
+                        0.0F);
+                GL11.glPushMatrix();
+                terd.renderEngine = Minecraft.getMinecraft().renderEngine;
+                terd.preDrawBatch();
+                GL11.glColor4f((float) (this.COLOR_R / COLOR_R), (float) (this.COLOR_G / COLOR_G), (float) (this.COLOR_B / COLOR_B), (this.COLOR_A / COLOR_A));
+                terd.renderTileEntity(te, partialTicks, -1);
+                terd.drawBatch(pass);
+                GL11.glPopMatrix();
+            }
+        }
+    }
+
+    private void renderGhostBlock(final World world, ModelHolder holder, final EntityPlayer player, BlockRenderLayer layer, boolean existingModel, final float partialTicks)
+    {
+        double dx = player.lastTickPosX + (player.posX - player.lastTickPosX) * partialTicks;
+        double dy = player.lastTickPosY + (player.posY - player.lastTickPosY) * partialTicks;
+        double dz = player.lastTickPosZ + (player.posZ - player.lastTickPosZ) * partialTicks;
+        BlockPos pos = holder.pos;
+
+        GlStateManager.pushMatrix();
+        GlStateManager.translate(pos.getX() - dx, pos.getY() - dy, pos.getZ() - dz);
+
+        if (existingModel)
+        {
+            GlStateManager.scale(1.001, 1.001, 1.001);
+        }
+
+        RenderHelper.disableStandardItemLighting();
+
+        if (layer == BlockRenderLayer.CUTOUT)
+        {
+            this.mc.getTextureManager().getTexture(TextureMap.LOCATION_BLOCKS_TEXTURE).setBlurMipmap(false, false);
+        }
+
+        GlStateManager.color(1f, 1f, 1f, 1f);
+
+        int alpha = ((int) (1.0F * 0xFF)) << 24;
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Use a "double" or "BigDecimal" instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2164)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Assign this magic number 24 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'alpha' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)